File: common/src/main/java/com/ssblur/scriptor/registry/colorable/ColorableBlockRegistry.java
package com.ssblur.scriptor.registry.colorable;

import com.ssblur.scriptor.color.interfaces.ColorableBlock;
import net.minecraft.world.level.block.Block;

import java.util.HashMap;

public class ColorableBlockRegistry {
  public static ColorableBlockRegistry INSTANCE = new ColorableBlockRegistry();

  HashMap<Block, ColorableBlock> colorableBlocks = new HashMap<>();

  public static DyeColorableBlocks DYE_COLORABLE_BLOCKS = new DyeColorableBlocks();

  public static void register(Block block, ColorableBlock colorableBlock) {
    INSTANCE.colorableBlocks.put(block, colorableBlock);
  }

  public static ColorableBlock get(Block block) {
    return INSTANCE.colorableBlocks.get(block);
  }

  public static boolean has(Block block) {
    return INSTANCE.colorableBlocks.containsKey(block);
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/registry/colorable/ColorableBlockRegistry.kt
package com.ssblur.scriptor.registry.colorable

import com.ssblur.scriptor.color.interfaces.ColorableBlock
import net.minecraft.world.level.block.Block

object ColorableBlockRegistry {
  var colorableBlocks: HashMap<Block, ColorableBlock> = HashMap()

  var DYE_COLORABLE_BLOCKS = DyeColorableBlocks

  fun register(block: Block, colorableBlock: ColorableBlock) {
    colorableBlocks[block] = colorableBlock
  }

  fun get(block: Block): ColorableBlock? {
    return colorableBlocks[block]
  }

  fun has(block: Block): Boolean {
    return colorableBlocks.containsKey(block)
  }
}
